src/Vegetation/photosynthesis.jl

```diff
@@ -192,3 +221,54 @@ function compute_photosynthesis(
     Rd = dark_respiration(Vcmax25, β, f, ΔHRd, T, To, R)
     return net_photosynthesis(Ac, Aj, Rd, β)
 end
+
+function compute_Jmax25(Jmax_model::PrescribedJmax25, _...)
+    return Jmax_model.Jmax25
+end
+
+
+function compute_Jmax25(Jmax_model::OptimalityJmax25, parameters::FarquharParameters, T, medlyn_factor,
+                         APAR, c_co2, R)
+    """Solve for Jmax25 assuming Aj=Ac"""
+    (;
+        Γstar25,
+        ΔHJmax,
+        ΔHVcmax,
+        ΔHΓstar,
+        f,
+        ΔHRd,
+        To,
+        θj,
+        ϕ,
+        mechanism,
+        sc,
+        ψc,
+        oi,
+        Kc25,
+        Ko25,
+        ΔHkc,
+        ΔHko,
+        jv_ratio,
+    ) = parameters
+
+    Γstar = co2_compensation(Γstar25, ΔHΓstar, T, To, R)
+    ci = intercellular_co2(c_co2, Γstar, medlyn_factor)
+    Kc = MM_Kc(Kc25, ΔHkc, T, To, R)
+    Ko = MM_Ko(Ko25, ΔHko, T, To, R)
+
+    # Light utilization of APAR
+    IPSII = ϕ * APAR / 2
+
+    Jmax_arr = arrhenius_function(T, To, R, ΔHJmax)
+    Vcmax_arr = arrhenius_function(T, To, R, ΔHVcmax)
+
+    if typeof(mechanism) == C3
+        C = Vcmax_arr/Jmax_arr/jv_ratio*(4*(ci + 2*Γstar)/(ci + Kc*(1 + oi/Ko)))
+    elseif typeof(mechanism) == C4
+        C = Vcmax_arr/Jmax_arr/jv_ratio
+    end
+
+    Jmax = (C - 1)*IPSII/(C*(C*θj - 1))
```

Comment:
Is this the inverse of https://github.com/CliMA/ClimaLSM.jl/blob/28e5083c1678192357b209cfc78fc832f156c132/src/Vegetation/canopy_parameterizations.jl#L193
?

Reply:
Almost, but not exactly. It is what you get if you set `Ac = Aj`, which for C3 means  `J * (ci - Γstar) / (4 * (ci + 2 * Γstar)) = Vcmax * (ci - Γstar) / (ci + Kc * (1 + oi / Ko))`. So `Jmax` is on both sides of the equation, in `J` on one side and implicitly in `Vcmax` on the other.